src/scripts/main.js

```diff
@@ -1,3 +1,323 @@
 'use strict';
 
-// write code here
+const body = document.querySelector('body');
+
+const table = document.querySelector('table');
+const tBody = table.querySelector('tbody');
+
+const message = document.createElement('div');
+const header = document.createElement('h2');
+const text = document.createElement('p');
+
+let rowsArray = Array.from(tBody.rows);
+
+let sortOrder;
+let sortHeader;
+
+table.addEventListener('click', function(ev) {
+  const elem = ev.target;
+
+  if (elem.tagName !== 'TH') {
+    return;
+  }
+
+  if (sortHeader !== elem) {
+    sortOrder = 'ASC';
+    sortHeader = elem;
+  } else {
+    sortOrder = 'DESC';
+    sortHeader = !sortHeader;
+  }
+
+  sortlist(elem.cellIndex, elem.dataset.type, sortOrder);
+});
+
+function sortlist(colNum, type, order) {
+  let compare;
+
+  if (order === 'ASC') {
+    compare = function(rowA, rowB) {
+      switch (type) {
+        case 'string':
+          return rowA.cells[colNum].textContent
+            .localeCompare(rowB.cells[colNum].textContent);
+
+        case 'number':
+          return rowA.cells[colNum].textContent
+          - rowB.cells[colNum].textContent;
+        case 'salary':
+          return toNumber(rowA.cells[colNum].textContent)
+          - toNumber(rowB.cells[colNum].textContent);
+      }
+    };
+  } else {
+    compare = function(rowA, rowB) {
+      switch (type) {
+        case 'string':
+          return rowB.cells[colNum].textContent
+            .localeCompare(rowA.cells[colNum].textContent);
+
+        case 'number':
+          return rowB.cells[colNum].textContent
+          - rowA.cells[colNum].textContent;
+        case 'salary':
+          return toNumber(rowB.cells[colNum].textContent)
+          - toNumber(rowA.cells[colNum].textContent);
+      }
+    };
+  }
```

Comment:
this code looks like duplicate. Try to optimize here. As I understood from your code, it will be enough to use 1 function and pass parameters to it in different order

```suggestion
  if (order === 'ASC') {
    compare = function(rowA, rowB)
  } else {
    compare = function(rowB, rowA)
  }
```

Reply:
I don't sure I changed it correctly. Can you tell me if now it's ok please?